ts/WoltLabSuite/Core/Ui/Search/Extended.ts

```diff
@@ -0,0 +1,197 @@
+      return;
+    }
+
+    this.updateQueryString();
+
+    if (this.lastSearchRequest) {
+      this.lastSearchRequest.abort();
+    }
+    const request = dboAction("search", "wcf\\data\\search\\SearchAction").payload(this.getFormData());
+    this.lastSearchRequest = request.getAbortController();
+    const { count, searchID, title, pages, template } = (await request.dispatch()) as ResponseSearch;
+    document.querySelector(".contentTitle")!.textContent = title;
+    this.searchID = searchID;
+    this.activePage = 1;
+
+    this.removeSearchResults();
+
+    if (count > 0) {
+      this.pages = pages!;
+      this.showSearchResults(template!);
+    }
+  }
+
+  private updateQueryString(): void {
+    const url = new URL(this.form.action);
+
+    new FormData(this.form).forEach((value, key) => {
+      if (value.toString()) {
+        url.search += url.search !== "" ? "&" : "?";
+        url.search += encodeURIComponent(key) + "=" + encodeURIComponent(value.toString());
+      }
+    });
+
+    window.history.replaceState({}, document.title, url.toString());
+  }
+
+  private getFormData(): Record<string, unknown> {
+    const data = {};
+    new FormData(this.form).forEach((value, key) => {
+      if (value.toString()) {
+        data[key] = value;
+      }
+    });
+
+    return data;
+  }
+
+  private initQueryString(): void {
+    const url = new URL(window.location.href);
+    url.searchParams.forEach((value, key) => {
+      let element = this.form.elements[key] as HTMLElement;
+      if (value && element) {
+        if (element instanceof RadioNodeList) {
+          let id = "";
+          element.forEach((childElement: HTMLElement) => {
+            if (childElement.classList.contains("inputDatePicker")) {
+              id = childElement.id;
+            }
+          });
+          if (id) {
+            DatePicker.setDate(id, new Date(value));
+          }
+        } else if (element instanceof HTMLInputElement) {
+          if (element.type === "checkbox") {
+            element.checked = true;
+          } else {
+            element.value = value;
+          }
+        } else if (element instanceof HTMLSelectElement) {
+          element.value = value;
+        }
+      }
+    });
+
+    this.typeInput.dispatchEvent(new Event("change"));
+    this.search();
+  }
+
+  private initPagination(position: "top" | "bottom"): void {
+    const wrapperDiv = document.createElement("div");
+    wrapperDiv.classList.add("pagination" + ucfirst(position));
+    this.form.parentElement!.appendChild(wrapperDiv);
+    const div = document.createElement("div");
+    wrapperDiv.appendChild(div);
+
+    new UiPagination(div, {
+      activePage: this.activePage,
+      maxPage: this.pages,
+
+      callbackSwitch: (pageNo) => {
+        this.changePage(pageNo);
+      },
+    });
+  }
+
+  private async changePage(pageNo: number): Promise<void> {
+    if (this.lastSearchResultRequest) {
+      this.lastSearchResultRequest.abort();
+    }
```

Comment:
I'd either explicitly clear the value after the abort:

```suggestion
    if (this.lastSearchResultRequest) {
      this.lastSearchResultRequest.abort();
      this.lastSearchRequest = undefined;
    }
```

 or just do:
```suggestion
    this.lastSearchResultRequest?.abort();
```